api/envoy/config/core/v3/protocol.proto

```diff
@@ -60,15 +60,23 @@ message UpstreamHttpProtocolOptions {
       "envoy.api.v2.core.UpstreamHttpProtocolOptions";
 
   // Set transport socket `SNI <https://en.wikipedia.org/wiki/Server_Name_Indication>`_ for new
-  // upstream connections based on the downstream HTTP host/authority header, as seen by the
-  // :ref:`router filter <config_http_filters_router>`.
+  // upstream connections based on the downstream HTTP host/authority header or any other arbitrary
+  // header when :ref:`alt_header_name <envoy_v3_api_field_config.core.v3.UpstreamHttpProtocolOptions.alt_header_name>`
+  // is set, as seen by the :ref:`router filter <config_http_filters_router>`.
   bool auto_sni = 1;
 
   // Automatic validate upstream presented certificate for new upstream connections based on the
-  // downstream HTTP host/authority header, as seen by the
-  // :ref:`router filter <config_http_filters_router>`.
-  // This field is intended to set with `auto_sni` field.
+  // downstream HTTP host/authority header or any other arbitrary header when :ref:`alt_header_name <envoy_v3_api_field_config.core.v3.UpstreamHttpProtocolOptions.alt_header_name>`
+  // is set, as seen by the :ref:`router filter <config_http_filters_router>`.
+  // This field is intended to be set with `auto_sni` field.
   bool auto_san_validation = 2;
+
+  // An optional alternative to the host/authority header to be used for setting the SNI value.
+  // It should be a valid downstream HTTP header, as seen by the
+  // :ref:`router filter <config_http_filters_router>`.
+  // If unset, host/authority header will be used for populating the SNI.
+  // This field is intended to be set with `auto_sni` field.
+  string alt_header_name = 3;
```

Comment:
```suggestion
  string alt_header_name = 3 [(validate.rules).string.well_known_regex = HTTP_HEADER_NAME];
```

Reply:
Thanks for the suggestion!

Reply:
@lizan Looks like setting this validation rule on this field makes it mandatory. Is there a way to keep it optional and apply this validation rule? All the tests which doesn't have this field defined are failing as the validation rule is failing.
```
[2021-09-09 06:18:13.999][3340][critical][main] [source/server/server.cc:114] error initializing configuration 'C:/botcode/w13/_tmp/22d16eec90a4c4a770d42be759b0bb77/bootstrap.pb': Proto constraint validation failed (HttpProtocolOptionsValidationError.UpstreamHttpProtocolOptions: embedded message failed validation | caused by UpstreamHttpProtocolOptionsValidationError.AltHeaderName: value does not match regex pattern "^:?[0-9a-zA-Z!#$%&'*+-.^_|~`]+$"): upstream_http_protocol_options {
  auto_sni: true
}
explicit_http_config {
  http3_protocol_options {
  }
}
```